src/main/java/ch/njol/skript/expressions/ExprMessage.java

```diff
@@ -1,225 +1,279 @@
+		CHAT("chat", "[chat( |-)]message", true) {
+			@Override
+			Class<? extends Event>[] getEvents() {
+				return new Class[]{PlayerChatEventHandler.usesAsyncEvent ? AsyncPlayerChatEvent.class : PlayerChatEvent.class};
+			}
+			
+			@Override
+			@Nullable
+			String get(final Event e) {
+				if (PlayerChatEventHandler.usesAsyncEvent)
+					return ((AsyncPlayerChatEvent) e).getMessage();
+				else
+					return ((PlayerChatEvent) e).getMessage();
+			}
+			
+			@Override
+			void set(final Event e, final String message) {
+				if (PlayerChatEventHandler.usesAsyncEvent)
+					((AsyncPlayerChatEvent) e).setMessage(message);
+				else
+					((PlayerChatEvent) e).setMessage(message);
+			}
+		},
+		JOIN("join", "(join|log[ ]in)( |-)message", true) {
+			@Override
+			Class<? extends Event>[] getEvents() {
+				return new Class[]{PlayerJoinEvent.class};
+			}
+			
+			@Override
+			@Nullable
+			String get(final Event e) {
+				return ((PlayerJoinEvent) e).getJoinMessage();
+			}
+			
+			@Override
+			void set(final Event e, final String message) {
+				((PlayerJoinEvent) e).setJoinMessage(message);
+			}
+		},
+		QUIT("quit", "(quit|leave|log[ ]out|kick)( |-)message", true) {
+			@Override
+			Class<? extends Event>[] getEvents() {
+				return new Class[]{PlayerQuitEvent.class, PlayerKickEvent.class};
+			}
+			
+			@Override
+			@Nullable
+			String get(final Event e) {
+				if (e instanceof PlayerKickEvent)
+					return ((PlayerKickEvent) e).getLeaveMessage();
+				else
+					return ((PlayerQuitEvent) e).getQuitMessage();
+			}
+			
+			@Override
+			void set(final Event e, final String message) {
+				if (e instanceof PlayerKickEvent)
+					((PlayerKickEvent) e).setLeaveMessage(message);
+				else
+					((PlayerQuitEvent) e).setQuitMessage(message);
+			}
+		},
+		DEATH("death", "death( |-)message", true) {
+			@Override
+			Class<? extends Event>[] getEvents() {
+				return new Class[]{EntityDeathEvent.class};
+			}
+			
+			@Override
+			@Nullable
+			String get(final Event e) {
+				if (e instanceof PlayerDeathEvent)
+					return ((PlayerDeathEvent) e).getDeathMessage();
+				return null;
+			}
+			
+			@Override
+			void set(Event e, String message) {
+			
+			}
+			
+		},
+		UNKNOWN_COMMAND("unknown command", "unknown (command|cmd) message", Skript.classExists("org.bukkit.event.command.UnknownCommandEvent")) {
+			@Override
+			Class<? extends Event>[] getEvents() {
+				return new Class[]{UnknownCommandEvent.class};
+			}
+			
+			@Nullable
+			@Override
+			String get(Event e) {
+				return ((UnknownCommandEvent) e).getMessage();
+			}
+			
+			@Override
+			void set(Event e, String message) {
+				((UnknownCommandEvent) e).setMessage(message);
+			}
+		};
+		
+		static ArrayList<String> patterns = new ArrayList<>();
+		static EnumSet<MessageType> supportedTypes;
+		
+		static {
+			MessageType[] types = values();
+			supportedTypes = EnumSet.allOf(MessageType.class);
+			for (int i = 0; i < supportedTypes.size(); i++) {
```

Comment:
calling size on a list you are modifying sounds like it might be problematic

Reply:
EnumSet's size is assured to be the number of enum constants so it shouldn't matter